src/main/java/com/squareup/javapoet/MethodSpec.java

```diff
@@ -181,26 +202,42 @@ public static Builder overriding(ExecutableElement method) {
     String methodName = method.getSimpleName().toString();
     MethodSpec.Builder methodBuilder = MethodSpec.methodBuilder(methodName);
 
-    // TODO copy method annotations.
-    // TODO check to ensure we're not duplicating override annotation.
     methodBuilder.addAnnotation(Override.class);
+    TypeMirror overrideType = elements.getTypeElement(Override.class.getCanonicalName()).asType();
+    for (AnnotationMirror mirror : method.getAnnotationMirrors()) {
+      if (types.isSameType(mirror.getAnnotationType(), overrideType)) {
+        continue;
+      }
+      methodBuilder.addAnnotation(AnnotationSpec.get(mirror));
+    }
 
-    modifiers = new LinkedHashSet<>(modifiers); // Local copy so we can remove.
+    modifiers = new LinkedHashSet<>(modifiers);
     modifiers.remove(Modifier.ABSTRACT);
     methodBuilder.addModifiers(modifiers);
 
     for (TypeParameterElement typeParameterElement : method.getTypeParameters()) {
-      methodBuilder.addTypeVariable(
-          TypeVariableName.get((TypeVariable) typeParameterElement.asType()));
-    }
-
-    methodBuilder.returns(TypeName.get(method.getReturnType()));
-
-    for (VariableElement parameter : method.getParameters()) {
-      // TODO copy parameter annotations.
-      methodBuilder.addParameter(TypeName.get(parameter.asType()),
-          parameter.getSimpleName().toString());
+      TypeVariable var = (TypeVariable) typeParameterElement.asType();
+      methodBuilder.addTypeVariable(TypeVariableName.get(var));
+    }
+
+    ExecutableType executableType = (ExecutableType) types.asMemberOf(containing, method);
+    methodBuilder.returns(TypeName.get(executableType.getReturnType()));
+
+    List<? extends VariableElement> parameters = method.getParameters();
+    List<? extends TypeMirror> parameterTypes = executableType.getParameterTypes();
+    for (int index = 0; index < parameters.size(); index++) {
+      VariableElement parameter = parameters.get(index);
+      TypeName type = TypeName.get(parameterTypes.get(index));
+      String name = parameter.getSimpleName().toString();
+      Modifier[] paramods = new Modifier[parameter.getModifiers().size()];
```

Comment:
True. Would help others as well. Wasn't there some similiar changes related to `Iterable` instead of varargs in this project ... or some other square project?